### ISSUE TITLE
get_aws_account_id Fails For Localstack

### ISSUE THREAD

**Issue by jonwinton** [gruntwork-io/terragrunt#1485]
Hi!

I'm using [Localstack](https://github.com/localstack/localstack) for local development testing and I'm running into issues where helper function calls to AWS fail at initialization time. Here's an example of the error:

```
Error in function call; Call to function "get_aws_account_id" failed: NoCredentialProviders: no valid providers in chain. Deprecated.
        For verbose messaging see aws.Config.CredentialsChainVerboseErrors.
```

If I remove these calls and hardcode the Localstack account id everything works properly and I can grab the account id from a Terraform `data` source (`data "aws_caller_identity" "current" {}`).

It looks like the built-in functions (like `get_aws_account_id`) are all evaluated using local credentials before any Terraform is evaluated, which means we can't set [a custom endpoint](https://docs.aws.amazon.com/sdk-for-go/v1/developer-guide/configuring-sdk.html#custom-endpoint) for for the session being used to grab values from AWS. [Here's the call to get the current aws account id](https://github.com/gruntwork-io/terragrunt/blob/master/aws_helper/config.go#L121).

I can't find any environment valiables to configure which endpoint to use for the AWS SDK, so it seems like this needs to be defined as a Terragrunt specific env var/runtime flag  that can be passed to the function. I'm happy to try and get a PR together, but I wanted to check that this would be good to implement.

Let me know if I missed something, and thanks in advance!

**Comment by brikis98** [comment]
You are correct! Terragrunt can't parse the Terraform code to find custom endpoints you've defined, as it may need to process something like `get_aws_account_id` _before_ it knows what Terraform code to even parse. So, the only option is to expose the ability to override SDK endpoints directly in Terragrunt itself. I don't see any way within the AWS Go SDK to override endpoints globally, so we'll have to do it when each config/client is being initialized. PR to support this is welcome.

**Comment by jonwinton** [comment]
@brikis98 thanks for confirming, I'll try and get a PR together soon 👍
